_drafts/conda.md

```diff
@@ -0,0 +1,716 @@
+  sha256: fc235ce853bf3bceba46eff4b95764c5935ca07fc4998762ef5e5b7d05f37085 # [osx]
+
+build:
+  string: "0"
+
+test:
+  commands:
+    - arm-none-eabi-gcc --version
+```
+
+- `package`: This is the Name and Version of the Conda package that we are
+  building. e.g. if we are building Make 4.2.1 for Conda, then Name would be
+  "Make" and Version would be "4.2.1".
+- `source`: This is either the fields `url` and `sha256` which points to a Zip
+  or Tarball containing the source code or binaries, or it can also reference a
+  Git repository using the `git_url` and `git_rev` fields.
+- `build`: This is the build identifier. If a user publishes a package but it
+  was built incorrectly, they would use a new version of this identifier when
+  they re-publish the package. A monotonically increasing integer or a random
+  hash is appropriate.
+- `test`: These are test commands to run against the package once it is build.
+  In our example above, we have the package print out the version.
+
+For information about all the fields in the `meta.yaml` file, reference
+[this documentation page](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html)
+
+### Writing the build script `build.sh`
+
+The last thing we need is to define a `build.sh` file. This script will be run
+after the build completes, and is responsible for building and moving the
+necessary binaries and supporting files to the directory that will be packaged
+up.
+
+If we download and extract the toolchain from the ARM website, the folder
+structure is as follows:
+
+```
+gcc-arm-none-eabi-8-2019-q3-update
+├── arm-none-eabi
+├── bin
+├── lib
+└── share
+```
+
+We'll want to copy **all** of these directories into our Conda package, so our
+script is simply
+
+```
+#!/bin/bash
+cp -R $SRC_DIR/* $PREFIX
+```
+
+`SRC_DIR` is the directory where the source files were extracted to, and
+`PREFIX` is the build prefix to which the build script should install the
+resulting files. For more variables exported during the build process, reference
+[this page](https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#environment-variables-set-during-the-build-process).
+
+### Building and Publishing the Package
+
+We've finished setting up our package metadata, so now it's time to build and
+publish it. To build the package, we run
+
+```
+(build)
+$ conda build .
+```
+
+This will download our source tarball from the ARM website, verify the checksum,
+extract the contents, and then, using the `build.sh` script, copy the contents
+to the package working directory, which it will then compress back into a
+tarball into a Conda package!
+
+After the build finishes, the command will output another command starting with
+`anaconda upload ...` which you can use to publish this new Conda package under
+your own personal repository.
+
+However, before publishing a package, I generally like to confirm it works in a
```

Comment:
You can add these as test commands in a conda recipe!
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-section

conda-build will create a blank environment to run the tests during the build process.

Reply:
Added a note about this. I used it in the real recipe but not the example.